### PR TITLE
Adding ctest for the new gmi converter

### PR DESCRIPTION
**Pull request by smaticka**
## Description

A ctest of the down-sampled `gmi` data needs to be added to confirm the `ioda-converter` works as it was designed. 

## Issue(s) addressed

Resolves #<issue_number>

## Checklist

- [x] add down-sampled raw data in `test/testinput`
- [x] add down-sampled converted data in `test/testoutput`
- [x] add the `ctest` information for `gmi` in `ioadconv/test/CMakeList.txt`
- [x] add converter to list which copies over `gmi_gpm.py` into the `build/bin/` directory.

### CODE REVIEW

**Comment by smaticka** [comment]
This PR is ready for review. I developed and tested on `orion`. The ctest passed after rebuilding.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -442,7 +444,7 @@ endforeach()
 # iodaconv_comp.sh test.
 #
 # For a converter that is generating new data, use a non-zero tolerance.
-set(IODA_CONV_COMP_TOL_ZERO "0.0")
+set(IODA_CONV_COMP_TOL_ZERO "1.0e-4")
```
Comment (PatNichols) [inline test/CMakeLists.txt:447]: If you set this below IODA_CONV_COMP_TOL  below does it still pass?

Comment (smaticka) [reply test/CMakeLists.txt]: @PatNichols Yes, it still passes. Is that the preferred order? If so, I'll adjust the `amsr2` converter as well.

Comment (PatNichols) [reply test/CMakeLists.txt]: @smaticka if still passes then use the smaller value.

**Review by BenjaminRuston — APPROVED** [review]
passing on my Mac with v1.4.0 stack

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -443,7 +445,7 @@ endforeach()
 #
 # For a converter that is generating new data, use a non-zero tolerance.
 set(IODA_CONV_COMP_TOL_ZERO "0.0")
-set(IODA_CONV_COMP_TOL "0.5e-4")
+set(IODA_CONV_COMP_TOL "1.0")
```
Comment (smaticka) [inline test/CMakeLists.txt:448]:
```suggestion
set(IODA_CONV_COMP_TOL "0.5e-4")
```

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -891,6 +893,18 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_sondetac
...
+                  TYPE    SCRIPT
+                  ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
+                  COMMAND bash
+                  ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                          netcdf
+                  "${CMAKE_BINARY_DIR}/bin/gmi_2ioda.py
+                  -i testinput/gmi_gpm_20220216.h5
+                  -o testrun/gmi_gpm_20220216.nc4
+                  -d 2022021600"
+                  gmi_gpm_20220216.nc4 ${IODA_CONV_COMP_TOL_ZERO})
```
Comment (smaticka) [inline test/CMakeLists.txt:906]:
```suggestion
                  gmi_gpm_20220216.nc4 ${IODA_CONV_COMP_TOL})
```

**Review by BenjaminRuston — APPROVED** [review]
yes this looks like what we want thanks @smaticka

**Review by fcvdb — APPROVED** [review]
Thanks Sam!

**Comment by smaticka** [comment]
@BenjaminRuston do you think the merging still needs to be blocked?

**Pull request merged**